nilearn/plotting/_utils.py

```diff
@@ -50,23 +51,31 @@ def check_surface_plotting_inputs(
 
     if isinstance(surf_map, SurfaceImage):
         if surf_mesh is None:
-            surf_mesh = surf_map.mesh.parts[hemi]
+            surf_mesh = _check_hemi_present(surf_map.mesh, hemi)
         if len(surf_map.shape) > 1 and surf_map.shape[1] > 1:
             raise TypeError(
                 "Input data has incompatible dimensionality. "
                 f"Expected dimension is ({surf_map.shape[0]},) "
                 f"or ({surf_map.shape[0]}, 1) "
                 f"and you provided a {surf_map.shape} surface image."
             )
-        surf_map = surf_map.data.parts[hemi].T
+        # concatenate the left and right data if hemi is "both"
+        if hemi == "both":
+            surf_map = np.concatenate(
```

Comment:
didn't you introduce a function to do that?

Reply:
That function concatenates the **mesh**.

Here, I am doing it for the data -- but this could be simplified (later?) by using `get_data` once we merge #4956